include/aws/mqtt/client.h

```diff
@@ -224,7 +255,10 @@ int aws_mqtt_client_connection_connect(
  *              otherwise AWS_OP_ERR and aws_last_error() is set.
  */
 AWS_MQTT_API
-int aws_mqtt_client_connection_disconnect(struct aws_mqtt_client_connection *connection);
+int aws_mqtt_client_connection_disconnect(
+    struct aws_mqtt_client_connection *connection,
+    aws_mqtt_client_on_disconnect_fn *on_disconnect,
+    void *userdata);
```

Comment:
We probably want disconnect to cancel any pending commands. Currently, that happens during destroy, but in python-land that's not going to happen until the connection is garbage-collected, which means basically never.